net/netns/netns_windows.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package netns
+
+import (
+	"encoding/binary"
+	"syscall"
+	"unsafe"
+
+	"github.com/tailscale/winipcfg-go"
+	"golang.org/x/sys/windows"
+)
+
+// control binds c to the Windows interface that holds a default
+// route, and is not the Tailscale WinTun interface.
+func control(network, address string, c syscall.RawConn) (err error) {
+	canV4, canV6 := false, false
+	switch network {
+	case "tcp", "udp":
+		canV4, canV6 = true, true
+	case "tcp4", "udp4":
+		canV4 = true
+	case "tcp6", "udp6":
+		canV6 = true
+	}
+
+	if canV4 {
+		if4, err := getDefaultInterface(winipcfg.AF_INET)
+		if err != nil {
+			return err
+		}
+		if err := bindSocket4(c, if4); err != nil {
+			return err
+		}
+	}
+
+	if canV6 {
+		if6, err := getDefaultInterface(winipcfg.AF_INET6)
+		if err != nil {
+			return err
+		}
+		if err := bindSocket6(c, if6); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getDefaultInterface returns the index of the interface that has the
+// non-Tailscale default route for the given address family.
+func getDefaultInterface(family winipcfg.AddressFamily) (ifidx uint32, err error) {
+	ifs, err := winipcfg.GetInterfaces()
+	if err != nil {
+		return 0, err
+	}
+	tsLUIDs := map[uint64]bool{}
+	for _, iface := range ifs {
+		if iface.Description == "Tailscale Tunnel" {
+			tsLUIDs[iface.Luid] = true
+		}
+	}
+
+	routes, err := winipcfg.GetRoutes(family)
+	if err != nil {
+		return 0, err
+	}
+
+	bestMetric := ^uint32(0)
+	// Zero is "unspecified". If we end up passing zero to
```

Comment:
A zero index or metric? Given the placement it's not exactly clear. Reword a bit?